File: array/max_Profit.py
prices = list(map(int, input().split()))
profit = 0 #初始化利润
buy = prices[0] #初始化买入价钱
for sell in prices:
    buy = min(sell, buy) #买入要最小
    profit = max(profit, sell-buy) #只记录最大利润
print(profit)

'''
leecode写法
    buy = prices[0]
    profit = 0
    for sell in prices:
        buy = min(buy, sell)
        profit = max(profit, sell-buy)
    return profit
'''

File: array/maxProfit.cpp
#include <iostream>
#include <vector>
int maxProfit(std::vector<int> prices)
{
    int res = 0;
    int in = prices[0];
    for (int i = 1; i < prices.size(); ++i)
    {
        int out = prices[i];
        in = std::min(in, out);
        res = std::max(res, out-in);
    }
    return res;
}

int main()
{
    std::vector<int> prices = {7,1,5,3,6,4};
    std::cout << maxProfit(prices);
}